test/boostTest.cpp

```diff
@@ -86,14 +86,15 @@ int registerTests(
 			fs::directory_iterator(fullpath),
 			fs::directory_iterator()
 		))
-			if (fs::is_directory(entry.path()) || TestCase::isTestFilename(entry.path().filename()))
+		{
+			bool addTest = true;
+			for (auto const& element: entry.path())
+				if (boost::starts_with(element.string(), "_"))
```

Comment:
Not really sure about this. E.g. if someone checks out the whole solidity  repo in a directory that is placed in a directory starting `_`, all tests would be ignored. I could somehow only check the directories starting from something like `test/libsolidity/semanticTests/`. I guess that would be cleaner.

Reply:
This is interesting. What about maybe adding a comment as a first line in a file that states this is external source file, and then check based on that?

Reply:
Initially I also added a special comment for external sources. The problem is that we need to change the external sources, if we want to use them here. So @axic suggested to use `_` as a prefix. I think thats a nice solution and we don't need to change any external sources, before we can use them in our test framework.

Reply:
Ok, sounds good

Reply:
Maybe we should actually implement like you suggested: check the directories starting from something like `test/libsolidity/semanticTests/`?

Reply:
Yeah, I think that would be better.

Reply:
I add a simple check for `test/libsolidity/semanticTests/`.